doc/blog/201809-UnoUnderTheHoodDependencyObjectGeneration.md

```diff
@@ -0,0 +1,76 @@
+# Talkin' 'bout my generation: How the Uno Platform generates code, part 2
+
+[Previously](https://medium.com/@unoplatform/talkin-bout-my-generation-how-the-uno-platform-generates-code-part-1-under-the-hood-7664d83c4f90) we looked at how the [Uno Platform](https://platform.uno/) turns XAML mark-up files into C# code.  In this article, I'll talk about another way Uno uses code generation, allowing us to make native Android and iOS views conform to UWP's API, and tackle the thorny problem of [multiple inheritance](https://en.wikipedia.org/wiki/Multiple_inheritance). 
+ 
+## Wanting it all
+Part of the power of Uno on Android and iOS comes from the ability to mix view types from the UWP framework with purely native views. But this poses a challenge for reproducing UWP's inheritance hierarchy: as I mentioned in an [earlier article](https://hackernoon.com/pushing-the-right-buttons-how-uno-implements-views-under-the-hood-a5e93ea86688), `UIElement` is the primitive view type in UWP, but it in turn derives from the `DependencyObject` class. `DependencyObject` exposes methods related to the dependency property/databinding system. 
+ 
+Multiple inheritance isn't permitted in C#, so in order for UIElement to be a `ViewGroup` on Android and a `UIView` on iOS, and still 'be' a `DependencyObject` as well, we opted to make `DependencyObject` an interface. 
+ 
+But that alone doesn't solve our problems. What if consumer code wants to inherit directly from `DependencyObject`? The following is completely valid in an Uno/UWP app:
+
+```` csharp
+public MyBindableObject : DependencyObject { 
+    // Custom dependency properties... 
+}
+````
+ 
+There are non-view framework types, too, which inherit from `DependencyObject`, like `Transforms` and `Brushes`. 
+ 
+We face a weaker form of this problem - wanting to have two base types - in other cases is well. In a few places in the framework we inherit from a more derived native view type. For example, `ScrollContentPresenter` inherits from the native scroll view on Android and iOS. But we also want `ScrollContentPresenter` to expose the methods and properties of `FrameworkElement`. 
+ 
+## Mixing it up 
```

Comment:
as discussed, before or after this subtitle I would mention that we successfully solved that problem using code gen.